pkg/client/pods.go

```diff
@@ -73,6 +74,11 @@ func (c *pods) Delete(name string) error {
 	return c.r.Delete().Namespace(c.ns).Resource("pods").Name(name).Do().Error()
 }
 
+// Same as above but with pod termination grace period specified.
+func (c *pods) DeleteGracefully(name string, gracePeriod uint64) error {
```

Comment:
It would be converted to query parameters.  We have decoding of query parameters into objects for api.DeleteOptions, api.ListOptions, etc.  This is the first place where a client can now encode.  You'll be encoding to map[string][]string, and then you'll add those to the query of the client/request.

----- Original Message -----
> > @@ -73,6 +74,11 @@ func (c *pods) Delete(name string) error {
> >  	return
> >  	c.r.Delete().Namespace(c.ns).Resource("pods").Name(name).Do().Error()
> >  }
> >  
> > +// Same as above but with pod termination grace period specified.
> > +func (c *pods) DeleteGracefully(name string, gracePeriod uint64) error {
> 
> You mean converting DeleteOptions object to appropriate version, encode it
> and attach as a body to the request, do I understand correctly?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/7432/files#r29275715